### PR TITLE
Picture database on firebase

### PR DESCRIPTION
**Pull request by LucaCarroz**
Added features to the FirebaseDB class:
-Store and Load images from the database, stored under a certain userId (this interface might change in the future, especially once google authentification will be merged)

### CODE REVIEW

**Review by DSchroeter96 — CHANGES_REQUESTED** [review]
Overall good first implementation of the Firebase utilization. We are able to store and retrieve pictures from the database, which was the requirement.

But I think it would be better to have a look at the suggestions before merging.

File: app/src/main/java/com/github/sdp_begreen/begreen/FirebaseDB.kt
Comment (LucaCarroz) [reply app/src/main/java/com/github/sdp_begreen/begreen/FirebaseDB.kt:65]: That's a good point, there are indeed other functionalities provided by Firebase that are more suited to store images. Since this (hopefully) will not change the interface of the FirebaseDB class, I'll make this change later, since Bryan is waiting for this merge advance. Thanks for pointing that out :)

File: app/src/main/java/com/github/sdp_begreen/begreen/FirebaseDB.kt
Comment (LucaCarroz) [reply app/src/main/java/com/github/sdp_begreen/begreen/FirebaseDB.kt:37]: Very good idea! Kotlin coroutines seem to be way better to use than Java's CompletableFutures and produce much clearer code apparently. Thanks a lot for your implementation :P

Comment (LucaCarroz) [reply app/src/main/java/com/github/sdp_begreen/begreen/FirebaseDB.kt:37]: Thanks a lot! I didnt know about Kotlin coroutines and from the documentation I just read it seems like a much better choice, I changed the "get" and "getImage" functions to "suspend" functions as you proposed. Thanks a lot for the code snipet :)

File: app/src/main/java/com/github/sdp_begreen/begreen/FirebaseDB.kt
Comment (LucaCarroz) [reply app/src/main/java/com/github/sdp_begreen/begreen/FirebaseDB.kt:65]: Yes indeed, storing images in the realtime database seems (and is) painful, I just thought I was meant to use that and not firestore or anything else. Since this should not modify the interface of the "FirebaseDB" methods, I'll keep it that way for the moment and change later since Bryan needs the implementation of "FirebaseDB" for one of his task, and he's waiting on me. I'll definitly apply changes soon

File: app/src/main/java/com/github/sdp_begreen/begreen/activities/DatabaseActivity.kt
Comment (LucaCarroz) [reply app/src/main/java/com/github/sdp_begreen/begreen/activities/DatabaseActivity.kt:34]: Indeed, thanks!

File: app/build.gradle
Comment (DSchroeter96) [inline app/build.gradle:68]: I think this import is not useful, coroutines seems to already be available in android environment without it.

File: app/src/androidTest/java/com/github/sdp_begreen/begreen/activities/DatabaseActivityTest.kt
Comment (DSchroeter96) [inline app/src/androidTest/java/com/github/sdp_begreen/begreen/activities/DatabaseActivityTest.kt:28]: Here I think it would be better to use the firebase emulator to test our database implementation, it will avoid polluting production database each time we tests the app.

```suggestion
        @Before
            fun init() {
                Firebase.database.useEmulator("10.0.2.2",9000)
            }
```

File: app/src/main/java/com/github/sdp_begreen/begreen/FirebaseDB.kt
Comment (DSchroeter96) [inline app/src/main/java/com/github/sdp_begreen/begreen/FirebaseDB.kt:31]: As we are using the emulator for test, I think it is unnecessary to keep the possibility to override the reference during test, so I would change the reference to a `private val`

**Review by DSchroeter96 — CHANGES_REQUESTED** [review]
The modifications were very good, good implementation of the coroutine. It makes the code much nicer.

However, I would still change the way we are doing test, to use the firebase emulator.

File: app/build.gradle
Comment (LucaCarroz) [reply app/build.gradle:68]: Indeed, I'll delete this import, thanks!

File: app/src/androidTest/java/com/github/sdp_begreen/begreen/activities/DatabaseActivityTest.kt
Comment (LucaCarroz) [reply app/src/androidTest/java/com/github/sdp_begreen/begreen/activities/DatabaseActivityTest.kt:28]: Indeed, the Emulator seemed to be painful to set up but it looks like it wasnt that much, thanks a lot!

**Review by DSchroeter96 — APPROVED** [review]
Great improvement, well done for your work !

File: app/src/main/java/com/github/sdp_begreen/begreen/FirebaseDB.kt
Comment (LucaCarroz) [reply app/src/main/java/com/github/sdp_begreen/begreen/FirebaseDB.kt:35]: Actually yes, thinking again about it, it might not be the best idea. If we're making a query to Firebase's database, we would like to know whether we got no value because no key was found of because we lost connection. I'll create a new branch to add this fix, aswell as some other changes in the way we store images since it is not optimal for the moment

**Pull request merged**
